src/solver/type_getters.jl

```diff
@@ -502,9 +509,54 @@ function get_sim_info(config::AtmosConfig)
         style = allowed_dir_styles[lowercase(requested_style)],
     )
 
+    restart_file = nothing
+
+    if parsed_args["detect_restart_file"]
+        lowercase(parsed_args["output_dir_style"]) == "activelink" ||
+            error("detect_restart_file works only with ActiveLink")
+        # output_dir will be something like ......./output_1234
+        name_rx = r"output_(\d\d\d\d)"
+        restart_file_rx = r"day\d+\.\w+\.hdf5"
+        counter_match = match(name_rx, output_dir)
+        if !isnothing(counter_match)
+            counter = parse(Int, counter_match[1])
+            if counter != 0
+                # As implemented in ClimaUtilities.OutputPathGenerator
```

Comment:
Where is this implemented in ClimaUtilities? If it's not reusable, could we make it reuseable to reduce code duplication?

Reply:
It's implemented in `ClimaUtilities.OutputPathGenerator`, but it is inside a function that does other things.

I can easily move it out and call it from ClimaAtmos to avoid duplication

Reply:
I realized I could make something more robust, so this is no longer needed/used.